File: EcoBookAiBackend/src/main/java/com/ecobook/dto/CreateMaterialRequestDTO.java
package com.ecobook.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class CreateMaterialRequestDTO {
    private String uploadId;
    private String titulo;
    private String descricao;
    private String disciplina;
    private String nivelEnsino;
    private Integer ano;
    private String sistemaEnsino;
    private String estadoConservacao;
    private Integer dataPublicacao;
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/CreateMaterialRequestDTO.kt
package com.ecobook.dto

data class CreateMaterialRequestDTO(
    val uploadId: String,
    val titulo: String,
    val autor: String?,
    val editora: String?,
    val descricao: String,
    val disciplina: String,
    val nivelEnsino: String,
    val ano: Int?,
    val sistemaEnsino: String,
    val estadoConservacao: String,
    val dataPublicacao: Int?
)
